src/charm.py

```diff
@@ -144,11 +152,13 @@ def _reconfigure(self, event: ops.charm.RelationEvent) -> None:
                 # update the set of replica set hosts
                 self._peers.data[self.app]["replica_set_hosts"] = json.dumps(self._unit_ips)
                 logger.debug("Replica set successfully reconfigured")
-                self.unit.status = ActiveStatus()
             except (ConnectionFailure, ConfigurationError, OperationFailure) as e:
                 logger.error("deferring reconfigure of replica set: %s", str(e))
                 self.unit.status = WaitingStatus("waiting to reconfigure replica set")
                 event.defer()
+                return
+
+        self.unit.status = ActiveStatus()
```

Comment:
nit: i usually try to avoid nested code structure if at all possible. in the above, would the following be better?

```
if not self._need_replica_set_reconfiguration:
    self.unit.status = ActiveStatus()
    return

try:
...
except:
...
```

Reply:
Yes! Much better, fully agree thanks for this suggestion Shayan